File: libs/sdk-java/src/main/java/io/daytona/sdk/model/SessionExecuteRequest.java
// Copyright Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

package io.daytona.sdk.model;

public class SessionExecuteRequest extends io.daytona.toolbox.client.model.SessionExecuteRequest {
    public SessionExecuteRequest() {}

    public SessionExecuteRequest(String command, Boolean runAsync) {
        super();
        setCommand(command);
        setRunAsync(runAsync);
    }

    public SessionExecuteRequest(io.daytona.toolbox.client.model.SessionExecuteRequest source) {
        super();
        if (source != null) {
            setCommand(source.getCommand());
            setRunAsync(source.getRunAsync());
        }
    }
}

File: libs/sdk-dotnet/Models/SessionExecuteRequest.cs
// Copyright Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

using System.Text.Json.Serialization;

namespace Daytona.Sdk.Models;

public record SessionExecuteRequest
{
    [JsonPropertyName("command")] public string Command { get; init; } = string.Empty;
    [JsonPropertyName("runAsync")] public bool? RunAsync { get; init; }
    [JsonPropertyName("suppressInputEcho")] public bool? SuppressInputEcho { get; init; }
}